cmd/create/testoperatorrelease/requests.go

```diff
@@ -0,0 +1,226 @@
+package testoperatorrelease
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"sigs.k8s.io/yaml"
+
+	"github.com/giantswarm/apiextensions/v2/pkg/apis/release/v1alpha1"
+	"github.com/giantswarm/microerror"
+)
+
+type apprequest struct {
+	Name    string `json:"name"`
+	Version string `json:"version"`
+	Issue   string `json:"issue"`
+}
+
+type request struct {
+	Name     string       `json:"name"`
+	Requests []apprequest `json:"requests"`
+}
+
+type requests struct {
+	Releases []request `json:"releases"`
+}
+
+func (r *runner) updateFromRequests(ctx context.Context, release *v1alpha1.Release) error {
+	reqs := requests{}
+	// Parse requests from the requests.yaml file.
+	{
+		requestsPath := filepath.Join(r.flag.ReleasesPath, r.flag.Provider, "requests.yaml")
+
+		releaseYAML, err := ioutil.ReadFile(requestsPath)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		err = yaml.Unmarshal(releaseYAML, &reqs)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	targetVersion, err := semver.NewVersion(strings.TrimPrefix(release.Name, "v"))
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.Debugf(ctx, "Looking for requests for version %v", targetVersion.String())
+
+	apps, err := r.mergeRequirements(ctx, reqs, *targetVersion)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	for appname, minVersionStr := range apps {
+		// Search app or component in the release CR.
+		found := false
+		for i, comp := range release.Spec.Components {
+			if comp.Name == appname {
+				found = true
+				currentVersion, err := semver.NewVersion(comp.Version)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				minVersion, err := semver.NewVersion(minVersionStr)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				if currentVersion.LessThan(minVersion) {
+					release.Spec.Components[i].Version = minVersion.String()
+				}
+				break
+			}
+		}
+
+		if found {
+			continue
+		}
+
+		for i, app := range release.Spec.Apps {
+			if app.Name == appname {
+				found = true
+				currentVersion, err := semver.NewVersion(app.Version)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				minVersion, err := semver.NewVersion(minVersionStr)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				if currentVersion.LessThan(minVersion) {
+					release.Spec.Apps[i].Version = minVersion.String()
+				}
+				break
+			}
+		}
+
+		if !found {
+			r.logger.LogCtx(ctx, "level", "warning", "message", fmt.Sprintf("Adding a new component from the requirements is not currently supported. Tried to add %s", appname))
+		}
+	}
+
+	return nil
+}
+
+func affects(r request, targetVersion semver.Version) (bool, error) {
```

Comment:
Some style comment. Instead of passing the whole `request`, I was thinking about passing only the request name as a parameter called `range` or something like that. And the function would be called `includes` / `within`? It feels clearer to me something like `includes(range, targetVersion)`. What do you think?

Reply:
Wooo! Nice let me test it

Reply:
I could have saved ~ 1.5 hours of development. Thanks Tuomas!